front/lib/api/assistant/preprocessing.ts

```diff
@@ -178,20 +178,43 @@ export async function renderConversationForModel(
         }
       );
 
-      let systemContext = "";
+      const metadataItems: string[] = [];
+
+      const identityTokens: string[] = [];
+      if (m.context.fullName) {
+        identityTokens.push(m.context.fullName);
+      }
+      if (m.context.username) {
+        const usernameToken = m.context.fullName
+          ? `(@${m.context.username})`
+          : `@${m.context.username}`;
+        identityTokens.push(usernameToken);
+      }
+      if (m.context.email) {
+        identityTokens.push(`<${m.context.email}>`);
+      }
+      if (identityTokens.length > 0) {
+        metadataItems.push(`- Sender: ${identityTokens.join(" ")}`);
+      }
+
+      if (m.created) {
+        metadataItems.push(`- Sent at: ${new Date(m.created).toISOString()}`);
```

Comment:
We should use the user locale string (with timezone) we're supposed to have it always I believe it's a required parameter of the API for that reason right?